exec/java-exec/src/test/java/org/apache/drill/TestPartitionPruning.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill;
+
+import org.apache.drill.categories.PlannerTest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.nio.file.Paths;
+
+@Category({PlannerTest.class})
+public class TestPartitionPruning extends PlanTestBase {
+
+  private static final String TABLE_WITH_METADATA = "parquetTable1";
+
+  @BeforeClass
+  public static void setUp() {
+    dirTestWatcher.copyResourceToRoot(Paths.get("multilevel"));
+    dirTestWatcher.copyResourceToRoot(Paths.get("multilevel/parquet"), Paths.get(TABLE_WITH_METADATA));
+  }
+
+  @Test
+  public void testPartitionPruning() throws Exception {
+    String[] tableNames = {"multilevel/parquet", "multilevel/json", "multilevel/csv"};
```

Comment:
Interesting that we had not Partition Pruning for the CSV or JSON formats earlier.
Could elaborate how to update Drill doc ["Partition Pruning Introduction"](https://drill.apache.org/docs/partition-pruning-introduction/) in accordance to your improvements?